include/fmt/chrono.h

```diff
@@ -2144,39 +2174,79 @@ template <typename Char> struct formatter<month, Char> {
       ++begin;
       localized = true;
     }
-    return begin;
+    use_tm_formatter_ = !localized && (begin != nullptr);
+    return use_tm_formatter_ ? formatter<std::tm, Char>::parse(ctx) : begin;
   }
 
   template <typename FormatContext>
   auto format(month m, FormatContext& ctx) const -> decltype(ctx.out()) {
     auto time = std::tm();
-    // std::chrono::month has a range of 1-12, std::tm requires 0-11
     time.tm_mon = static_cast<int>(static_cast<unsigned>(m)) - 1;
+    if (use_tm_formatter_) {
+      return formatter<std::tm, Char>::format(time, ctx);
+    }
     detail::get_locale loc(localized, ctx.locale());
     auto w = detail::tm_writer<decltype(ctx.out()), Char>(loc, ctx.out(), time);
     w.on_abbr_month();
     return w.out();
   }
 };
 
-template <typename Char> struct formatter<year, Char> {
+template <typename Char>
+struct formatter<year, Char> : formatter<std::tm, Char> {
+ private:
+  bool use_tm_formatter_{false};
+
+ public:
   FMT_CONSTEXPR auto parse(basic_format_parse_context<Char>& ctx)
       -> decltype(ctx.begin()) {
-    return ctx.begin();
+    use_tm_formatter_ = ctx.begin() != nullptr;
+    return formatter<std::tm, Char>::parse(ctx);
   }
 
   template <typename FormatContext>
   auto format(year y, FormatContext& ctx) const -> decltype(ctx.out()) {
     auto time = std::tm();
-    // std::tm::tm_year is years since 1900
     time.tm_year = static_cast<int>(y) - 1900;
-    detail::get_locale loc(true, ctx.locale());
+    if (use_tm_formatter_) {
+      return formatter<std::tm, Char>::format(time, ctx);
+    }
+    detail::get_locale loc(false, ctx.locale());
     auto w = detail::tm_writer<decltype(ctx.out()), Char>(loc, ctx.out(), time);
     w.on_year(detail::numeric_system::standard);
     return w.out();
   }
 };
 
+template <typename Char>
+struct formatter<year_month_day, Char> : formatter<std::tm, Char> {
+ private:
+  bool use_tm_formatter_{false};
+
+ public:
+  FMT_CONSTEXPR auto parse(basic_format_parse_context<Char>& ctx)
+      -> decltype(ctx.begin()) {
+    use_tm_formatter_ = ctx.begin() != nullptr;
```

Comment:
I think, if you want to check that format is not empty, then it is more correct to use logic:

https://github.com/fmtlib/fmt/blob/3b5f3de3b57505c7f1a60ee40ef3448c623b1326/include/fmt/chrono.h#L2351-L2352

Code from snippet is checks that format is empty.